### PR TITLE
JWT: Set `private` directive to the CC header

### PR DESCRIPTION
**Pull request by alex-schneider**
Automatically add the `private` directive to the response `Cache-Control` HTTP header field value for all resources protected by JWT.

---
<details>
    <summary>Reviewer checklist</summary>
    <ul>
        <li>Read PR description: a summary about the changes is required</li>
        <li>Changelog updated</li>
        <li>Documentation: docs/{Reference, Cli, ...}, Docker and cli help/usage</li>
        <li>Pulled branch, manually tested</li>
        <li>Verified requirements are met</li>
        <li>Reviewed the code</li>
        <li>Reviewed the related tests</li>
    </ul>
</details>

### CODE REVIEW

File: server/http_endpoints_test.go [hunk 1]
```diff
@@ -823,6 +827,14 @@ func TestEndpointACBufferOptions(t *testing.T) {
 				st.Errorf("want: %d, got: %d", tc.expectedStatus, res.StatusCode)
 			}
 
+			// Prevent diff errors from an unordered list.
```
Comment (johakoch) [inline server/http_endpoints_test.go:828]: Better create an _additional_ test (e.g. in http_integration_test.go, where several JWT tests already reside) than "hijack" the test named TestEndpointACBufferOptions to check for something very difference.

**Review by johakoch — CHANGES_REQUESTED** [review]

File: docs/REFERENCE.md [hunk 2]
```diff
@@ -379,6 +379,7 @@ required _label_.
 | `jwks_url` | string | - | URI pointing to a set of [JSON Web Keys (RFC 7517)](https://datatracker.ietf.org/doc/html/rfc7517) | - | `jwks_url = "http://identityprovider:8080/jwks.json"` |
 | `jwks_ttl` | [duration](#duration) | `"1h"` | Time period the JWK set stays valid and may be cached. | - | `jwks_ttl = "1800s"` |
 | `backend`  | string| - | [backend reference](#backend-block) for enhancing JWKS requests| - | `backend = "jwks_backend"` |
+| `disable_private_caching` | bool | `false` | If set to `true`, Couper does not add the `private` directive to the `Cache-Control` HTTP header field value. | - | - |
```
Comment (johakoch) [inline docs/REFERENCE.md:382]: The new default behaviour (adding `private`) should be documented in prose in the section on JWT; not only the possibility to change this default using this attribute.

File: config/runtime/server.go [hunk 3]
```diff
@@ -637,15 +638,16 @@ func configureAccessControls(conf *config.Couper, confCtx *hcl.EvalContext, log
...
-					Algorithm:      jwtConf.SignatureAlgorithm,
-					Claims:         jwtConf.Claims,
-					ClaimsRequired: jwtConf.ClaimsRequired,
-					Key:            key,
-					Name:           jwtConf.Name,
-					RolesClaim:     jwtConf.RolesClaim,
-					RolesMap:       jwtConf.RolesMap,
-					ScopeClaim:     jwtConf.ScopeClaim,
-					Source:         ac.NewJWTSource(jwtConf.Cookie, jwtConf.Header, jwtConf.TokenValue),
+					Algorithm:             jwtConf.SignatureAlgorithm,
```
Comment (johakoch) [inline config/runtime/server.go:641]: possible improvement:
* create `&ac.JWTOptions{...}` with members used in **both** cases outside the `if`
* set the members specific to the case (`if` / `else`) and then call `ac.NewJWTFromJWKS()` / `ac.NewJWT()` with the modified `JWTOptions`

So there will be a lower probability that a new common member is added in only one of the cases

**Review by johakoch — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -10,6 +10,7 @@ Unreleased changes are available as `avenga/couper:edge` container.
   * [`jwt` block](./docs/REFERENCE.md#jwt-block)/[`jwt_signing_profile` block](./docs/REFERENCE.md#jwt-signing-profile-block) with ECDSA support ([#401](https://github.com/avenga/couper/issues/401))
   * Support for [sequences](./docs/REFERENCE.md#endpoint-sequence) of outgoing endpoint requests ([#405](https://github.com/avenga/couper/issues/405))
   * `expected_status` attribute for `request` and `proxy` block definitions which can be caught with [error handling](./docs/ERRORS.md#endpoint-related-error_handler) ([#405](https://github.com/avenga/couper/issues/405))
+  * `disable_private_caching` attribute for the [JWT Block](./docs/REFERENCE.md#jwt-block) ([#418](https://github.com/avenga/couper/pull/418))
```
Comment (malud) [inline CHANGELOG.md:13]: Changelog rebase required (-> unreleased)

**Review by malud — CHANGES_REQUESTED** [review]

**Review by malud — APPROVED** [review]

**Pull request merged**